golang/cosmos/x/swingset/keeper/store_snapshotter.go

```diff
@@ -0,0 +1,685 @@
+package keeper
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"regexp"
+
+	"github.com/Agoric/agoric-sdk/golang/cosmos/vm"
+	"github.com/Agoric/agoric-sdk/golang/cosmos/x/swingset/types"
+	vstoragetypes "github.com/Agoric/agoric-sdk/golang/cosmos/x/vstorage/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/tendermint/tendermint/libs/log"
+)
+
+// This module abstract the handling of swing-store snapshots, also known as
+// swing-store imports/exports, and the necessary communication with the
+// JS side.
```

Comment:
Do we have anywhere that describes how swing-store exports fit into the overall structure of cosmos state-sync snapshots? My recollection from our call is that there is some hook by which a module registers its integration(s) for cosmos to call when creating/loading/etc. a snapshot, and that the responsibility for creation is to produce an arbitrary number of protobuf "extension" entries with a common format version and for loading is to validate the format version of each extension entry and then consume it as appropriate.